src/test/java/org/junit/internal/runners/statements/ExpectExceptionTest.java

```diff
@@ -0,0 +1,81 @@
+
+import org.junit.Test;
+import org.junit.internal.AssumptionViolatedException;
+import org.junit.runners.model.Statement;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+/**
+ * Integration tests can be found in {@link org.junit.tests.running.methods.ExpectedTest}.
+ * See e.g. {@link org.junit.tests.running.methods.ExpectedTest#expectsAssumptionViolatedException()}
+ */
+public class ExpectExceptionTest {
+
+    @Test
+    public void whenExpectingAssumptionViolatedExceptionStatementsThrowingItShouldPass() {
+        Statement statementThrowingAssumptionViolatedException = new Fail(new AssumptionViolatedException("expected"));
+        ExpectException expectException = new ExpectException(statementThrowingAssumptionViolatedException, AssumptionViolatedException.class);
+
+        try {
+            expectException.evaluate();
+            // then no exception should be thrown
+        } catch (Throwable e) {
+            fail("should not throw anything, but was thrown: " + e);
+        }
+    }
+
+    @Test
+    public void whenExpectingAssumptionViolatedExceptionStatementsThrowingSubclassShouldPass() {
+        Statement statementThrowingAssumptionViolatedExceptionSubclass = new Fail(new org.junit.AssumptionViolatedException("expected"));
+        ExpectException expectException = new ExpectException(statementThrowingAssumptionViolatedExceptionSubclass, AssumptionViolatedException.class);
+
+        try {
+            expectException.evaluate();
+            // then no exception should be thrown
+        } catch (Throwable e) {
+            fail("should not throw anything, but was thrown: " + e);
+        }
+    }
+
+    @Test
+    public void whenExpectingAssumptionViolatedExceptionStatementsThrowingDifferentExceptionShouldFail() {
+        Statement statementThrowingSomeException = new Fail(new SomeException("not expected"));
+        ExpectException expectException = new ExpectException(statementThrowingSomeException, AssumptionViolatedException.class);
+
+        try {
+            expectException.evaluate();
+            fail("should throw 'Unexpected exception' when statement throws an exception which is not the one expected");
+        } catch (Exception e) {
+            assertThat(e.getMessage(), equalTo("Unexpected exception, expected<org.junit.internal.AssumptionViolatedException> " +
+                    "but was<org.junit.internal.runners.statements.ExpectExceptionTest$SomeException>"));
+        }
+    }
+
+    @Test
+    public void whenExpectingAssumptionViolatedExceptionStatementsPassingShouldFail() throws Exception {
+        ExpectException expectException = new ExpectException(new PassingStatement(), AssumptionViolatedException.class);
+
+        try {
+            expectException.evaluate();
+            fail("ExpectException should throw when the given statement passes");
```

Comment:
The idiom.I wrote above is quite common in JUnit's tests, so let's go with that. We can't simply rely on checking the exception  message because it also can result in tests passing that shouldn't pass, plus the failure messages you get if the test fails are confusing.

I don't think a new exception  type is worth it, bu the other maintainers might disagree with me and agree with you :-(

Verifying that a method or constructor call throws a specific exception would be easier if we could use lambdas and `assertThrows`